src/Frame/Frame.cc

```diff
@@ -1119,8 +1125,13 @@ bool Frame::FillSpatialProfileData(PointXy point, std::vector<CARTA::SetSpatialR
         } else {
             casacore::Slicer section = GetImageSlicer(AxisRange(x), AxisRange(y), AxisRange(CurrentZ()), stokes);
             const auto N = section.length().product();
+#ifdef HAS_SHARED_PTR_ARRAY
             std::shared_ptr<float[]> data(new float[N]); // zero initialization
             if (GetSlicerData(section, data.get())) {
+#else
+            std::vector<float> data(N);
```

Comment:
I am questioning myself now if we should try to keep the `shared_ptr` using:
```
std::shared_ptr<float> data(new float[N], [](int *v){
    delete[] v;
});
```
I could try if I configure a VM and make another pull request for that. However, this will work in the meantime and affects one version only.

Reply:
This seems like a good idea to me